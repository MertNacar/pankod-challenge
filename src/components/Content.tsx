import React, { FormEventHandler } from "react";
import SubNavBar from "./SubNavbar";
import SearchBox from "./SearchBox";
import SeriesMoviesList from "./SeriesMoviesList";
import $ from "jquery";

interface IProps {
  subTitle: string;
  entries: Array<Object>;
  dropdownEntries: Array<string>;
  searchValue: string;
  searchChange: FormEventHandler<HTMLInputElement>;
  searchClick: FormEventHandler<HTMLButtonElement>;
  searchPlaceholder: string;
  sendDropdownValue: (value: string) => void;
  dropdownValue: string;
}

const Content: React.FC<IProps> = props => {
  let dropdownItems = props.dropdownEntries.map((item: any, index: number) => {
    return (
      <button key={index} className="dropdown-item">
        {item}
      </button>
    );
  });

  //get value dropdown and send to parent
  function getValueDropdown() {
    $("#dropdownMovies button").on("click", function() {
      if ($(this).text() !== props.dropdownValue) {
        props.sendDropdownValue($(this).text());
      }
    });
  }

  return (
    <div>
      <SubNavBar title={props.subTitle} />
      <div className="m-5">
        <div
          className="d-flex flex-column flex-sm-row justify-content-between m-3"
          style={{ width: "90%" }}
        >
          <SearchBox
            searchPlaceholder={props.searchPlaceholder}
            searchValue={props.searchValue}
            searchChange={props.searchChange}
            searchClick={props.searchClick}
          />

          <div className="dropdown mt-sm-5">
            <button
              className="btn btn-light dropdown-toggle"
              type="button"
              id="orderButton"
              data-toggle="dropdown"
              onClick={() => getValueDropdown()}
            >
              {props.dropdownValue}
            </button>
            <div
              id="dropdownMovies"
              className="dropdown-menu dropdown-menu-right"
            >
              {dropdownItems}
            </div>
          </div>
        </div>
        <div className="d-flex justify-content-start flex-wrap mx-3">
          <SeriesMoviesList values={props.entries} classList="pr-3 pt-3 pb-3" />
        </div>
      </div>
    </div>
  );
};

export default Content;
