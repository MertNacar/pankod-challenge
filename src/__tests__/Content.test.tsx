import React from "react";
import { shallow } from "enzyme";
import Content from "../components/Content";
import mockData from "../../public/feed/test/mock_test.json";

describe("Content Component", () => {
  const wrapper = shallow(
    <Content
      subTitle="Popular Movies"
      entries={mockData}
      dropdownEntries={[
        "Year ascending",
        "Year descending",
        "Title ascending",
        "Title descending"
      ]}
      searchValue="Mocking"
      searchChange={() => jest.fn()}
      searchClick={() => jest.fn()}
      searchPlaceholder="Search..."
      sendDropdownValue={() => jest.fn()}
      dropdownValue="Title Ascending"
    />
  );

  it("is exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("return five items", () => {
    expect(wrapper).toMatchSnapshot("Content");
  });
});
