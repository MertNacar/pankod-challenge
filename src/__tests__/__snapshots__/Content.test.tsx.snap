// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component return five items: Content 1`] = `
<div>
  <SubNavBar
    title="Popular Movies"
  />
  <div
    className="m-5"
  >
    <div
      className="d-flex flex-column flex-sm-row justify-content-between m-3 h-100"
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <SearchBox
        searchChange={[Function]}
        searchClick={[Function]}
        searchPlaceholder="Search..."
        searchValue="Mocking"
      />
      <div
        className="dropdown mt-5 mt-sm-0"
      >
        <button
          className="btn btn-light dropdown-toggle"
          data-toggle="dropdown"
          id="orderButton"
          onClick={[Function]}
          type="button"
        >
          Title Ascending
        </button>
        <div
          className="dropdown-menu dropdown-menu-right"
          id="dropdownMovies"
        >
          <button
            className="dropdown-item"
            key="0"
          >
            Year ascending
          </button>
          <button
            className="dropdown-item"
            key="1"
          >
            Year descending
          </button>
          <button
            className="dropdown-item"
            key="2"
          >
            Title ascending
          </button>
          <button
            className="dropdown-item"
            key="3"
          >
            Title descending
          </button>
        </div>
      </div>
    </div>
    <div
      className="d-flex justify-content-start flex-wrap mx-3"
    >
      <SeriesMoviesList
        classList="pr-3 pt-3 pb-3"
        values={
          Array [
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              "images": Object {
                "Poster Art": Object {
                  "height": 1500,
                  "url": "https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg",
                  "width": 1000,
                },
              },
              "programType": "series",
              "releaseYear": 2016,
              "title": "Wolf Creek",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              "images": Object {
                "Poster Art": Object {
                  "height": 1500,
                  "url": "https://streamcoimg-a.akamaihd.net/000/106/36/10636-PosterArt-9add943c5e62c2d89f6d31458d33508a.jpg",
                  "width": 1000,
                },
              },
              "programType": "series",
              "releaseYear": 2015,
              "title": "No Activity",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              "images": Object {
                "Poster Art": Object {
                  "height": 1080,
                  "url": "https://streamcoimg-a.akamaihd.net/000/117/25/11725-PosterArt-deecf8dbd786dfa2d964413b0bf83726.jpg",
                  "width": 720,
                },
              },
              "programType": "series",
              "releaseYear": 2016,
              "title": "Billions",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              "images": Object {
                "Poster Art": Object {
                  "height": 1500,
                  "url": "https://streamcoimg-a.akamaihd.net/000/175/7/1757-PosterArt-fc0e2a5b18d4a662717f3cbb1827871d.jpg",
                  "width": 1000,
                },
              },
              "programType": "series",
              "releaseYear": 2015,
              "title": "Better Call Saul",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              "images": Object {
                "Poster Art": Object {
                  "height": 1500,
                  "url": "https://streamcoimg-a.akamaihd.net/000/143/24/14324-PosterArt-a0168953a3b93681dbd60c7bcc50a0af.jpg",
                  "width": 1000,
                },
              },
              "programType": "series",
              "releaseYear": 2015,
              "title": "Younger",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
